internal/app/go-filecoin/paymentchannel/testing/fake_paych_actor_util.go

```diff
@@ -24,43 +26,33 @@ import (
 // FakePaychActorUtil fulfils the MsgSender and MsgWaiter interfaces for a Manager
 // via the specs_actors mock runtime. It executes paych.Actor exports directly.
 type FakePaychActorUtil struct {
-	t   *testing.T
+	*testing.T
 	ctx context.Context
 	*mock.Runtime
 	*pcActorHarness
 	PaychAddr, PaychIDAddr, Client, ClientID, Miner address.Address
+	SendErr                                         error
 	result                                          MsgResult
 }
 
-// NewFakePaychActorUtil intializes a FakePaychActorUtil
-func NewFakePaychActorUtil(ctx context.Context, t *testing.T, paychBal abi.TokenAmount) *FakePaychActorUtil {
-	fai := &FakePaychActorUtil{
-		t:              t,
-		ctx:            ctx,
-		PaychAddr:      spect.NewActorAddr(t, "paychactor"),
-		PaychIDAddr:    spect.NewIDAddr(t, 999),
-		Client:         spect.NewActorAddr(t, "clientactor"),
-		ClientID:       spect.NewIDAddr(t, 980),
-		Miner:          spect.NewActorAddr(t, "mineractor"),
-		pcActorHarness: new(pcActorHarness),
-	}
-	fai.constructPaychActor(paychBal)
-	return fai
-}
+// ConstructPaychActor creates a mock.Runtime and constructs a payment channel harness + Actor
+func (fai *FakePaychActorUtil) ConstructPaychActor(t *testing.T, paychBal abi.TokenAmount) {
+	hasher := func(data []byte) [32]byte { return [32]byte{} }
 
-// constructPaychActor creates a mock.Runtime and constructs a payment channel harness + Actor
-func (fai *FakePaychActorUtil) constructPaychActor(paychBal abi.TokenAmount) {
 	builder := mock.NewBuilder(fai.ctx, fai.PaychAddr).
 		WithBalance(paychBal, big.Zero()).
 		WithEpoch(abi.ChainEpoch(42)).
 		WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID).
 		WithActorType(fai.PaychIDAddr, builtin.AccountActorCodeID).
```

Comment:
Blocking: A payment channel actor is not an account actor, so this is very misleading. I recently fixed a bunch of similar things in the specs-actors unit tests. In that case, it resulted from re-use of the paych actor as its own target. This is an invalid configuration, since a paych actor is not a signer and can never be a target. Please change this to PaymentChannelActorCodeID.